itests/kit/deals.go

```diff
@@ -87,6 +87,17 @@ func NewDealHarness(t *testing.T, client *TestFullNode, main *TestMiner, market
 //
 // TODO: convert input parameters to struct, and add size as an input param.
 func (dh *DealHarness) MakeOnlineDeal(ctx context.Context, params MakeFullDealParams) (deal *cid.Cid, res *api.ImportRes, path string) {
+	deal, res, path = dh.StartRandomDeal(ctx, params)
+
+	// TODO: this sleep is only necessary because deals don't immediately get logged in the dealstore, we should fix this
+	time.Sleep(time.Second)
+	fmt.Printf("WAIT DEAL SEALEDS START\n")
+	dh.WaitDealSealed(ctx, deal, false, false, nil)
```

Comment:
If the deal pipeline is buggy it appears this can hang forever.  So that possible flakes are easier to diagnose it would be nice to have a long timeout here (say 1 minutes) which fails if this doesn't finish waiting.

Reply:
Timeout probably belongs inside `WaitDealSealed`.